datalad/support/gitrepo.py

```diff
@@ -526,8 +526,15 @@ def git_get_remote_branches(self):
         # For some reason, this is three times faster than the version below:
         remote_branches = list()
         for remote in self.repo.remotes:
-            for ref in remote.refs:
-                remote_branches.append(ref.name)
+            try:
+                for ref in remote.refs:
+                    remote_branches.append(ref.name)
+            except AssertionError as e:
+                if e.message.endswith("did not have any references"):
+                    # this will happen with git annex special remotes
+                    pass
+                else:
+                    raise e
```

Comment:
and it is all kosher if 1 is annex special and others regular?

if only there was a test which should be real easy to create